modules/core/shared/src/main/scala/io/circe/Json.scala

```diff
@@ -547,7 +553,7 @@ object Json {
   /**
    * Create a `Json` value representing a JSON number from a `Long`.
    */
-  final def fromLong(value: Long): Json = JNumber(JsonLong(value))
+  final def fromLong(value: Long): Json = JNumber(JsonNumber.fromLong(value))
```

Comment:
If you change the table to `JNumber` as suggested in another commented then you would also change here:

```suggestion
  final def fromLong(value: Long): Json = JsonNumber.fromLong(value)
```